WcaOnRails/app/models/user.rb

```diff
@@ -626,6 +626,20 @@ def banned_at_date?(date)
     end
   end
 
+  private def can_edit_any_groups?
+    admin? || board_member? || results_team?
+  end
+
+  private def groups_with_edit_access
```

Comment:
The name implies that this is generic for any group. But in reality it only covers Delegate Regions.

In the `def permissions` method below, you also have cases for example for `can_edit_teams_committees`. In terms of code, these are groups as well, why not consider them?

Reply:
Currently, teams cannot be combined with this because teams doesn't have a groupId now. But once the migration of teams is completed, this function can be expanded to them as well. It can actually be expanded to other delegate group types that are already migrated like delegate_probations & translators, I'll do that later in another PR.